### DISCUSSION TITLE
Custom page with date filters and multiple widget cards

### DISCUSSION THREAD

**Discussion by johncarter-** [filamentphp/filament#3037] [Help]
Could someone provide me a top level overview of how I would achieve a layout like this?

![image](https://user-images.githubusercontent.com/3776888/177494513-0067f855-7fb6-4aef-9a3f-486e0bb78c5e.png)

I assume I need to make a custom page that uses `use Forms\Concerns\InteractsWithForms;` to enable the form fields at the top. Then somehow need to render the widget cards using `protected function getCards(): array`. 

Is that the right way? If so how do I render the `getCards()` in the view?

**Comment by danharrin** [comment]
You can avoid `getCards()` if you have a custom widget & view and literally just do this in Blade to render a card:

```php
{{ Card::make()... }}
```

**Reply by johncarter-** [reply]
I chose to do things this way and it was going really well, until I wanted to use a `DoughnutChartWidget` in the same way as a `\Filament\Widgets\StatsOverviewWidget\Card` - how do I go about that?

**Reply by danharrin** [reply]
Yeah, charts are different, as they are not extracted into Blade components like cards are. Maybe that should be a PR though, so you can reuse them outside of the Livewire context.

**Reply by johncarter-** [reply]
Gotcha.

**Comment by studiowizjo** [comment]
1. Create custom page with form
2. Inside form define fields with your filters. Each of them should be `reactive()` and call your method to query data 
3. Your widgets could be also part of the form- placed as [View](https://filamentphp.com/docs/2.x/forms/fields#view)  field

**Reply by johncarter-** [reply]
Thanks a lot for this, I wasn't quite sure to make use of the `Filament\Widgets\StatsOverviewWidget\Card` inside a custom `View` field so I am currently doing it using Dan's method above, but your way seems reasonable too as you don't need to do any custom HTML markup and you can reply on `Grid::make()->columns(9)`, I might go full circle and use this in the end if I need it.

**Reply by margarizaldi** [reply]
> 1. Create custom page with form
> ...

Hi, @studiowizjo , would you like to give me a little example of how to do that? At least until we can display a form component on the page. Thanks.

**Reply by studiowizjo** [reply]
```blade
<x-filament::page>

  <form class="space-y-6 filament-form" wire:submit.prevent="save">
    {{ $this->form }}

    <div class="flex flex-wrap items-center gap-4 filament-page-actions justify-start filament-form-actions">
      <x-filament::button type="submit" wire:loading.attr="disabled" hasLoadingIndicator="true" >
        <svg wire:loading="" wire:target="save" class="filament-button-icon w-6 h-6 mr-1 -ml-2 rtl:ml-1 rtl:-mr-2 animate-spin" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor">
          <path d="M2 12C2 6.47715 6.47715 2 12 2V5C8.13401 5 5 8.13401 5 12H2Z"></path>
        </svg>
        <span>Save</span>
      </x-filament::button>
    </div>

  </form>

</x-filament::page>
```

**Reply by margarizaldi** [reply]
> ```blade
> <x-filament::page>
> 
>   <form class="space-y-6 filament-form" wire:submit.prevent="save">
>     {{ $this->form }}
> 
>     <div class="flex flex-wrap items-center gap-4 filament-page-actions justify-start filament-form-actions">
>       <x-filament::button type="submit" wire:loading.attr="disabled" hasLoadingIndicator="true" >
>         <svg wire:loading="" wire:target="save" class="filament-button-icon w-6 h-6 mr-1 -ml-2 rtl:ml-1 rtl:-mr-2 animate-spin" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor">
>           <path d="M2 12C2 6.47715 6.47715 2 12 2V5C8.13401 5 5 8.13401 5 12H2Z"></path>
>         </svg>
>         <span>Save</span>
>       </x-filament::button>
>     </div>
> 
>   </form>
> 
> </x-filament::page>
> ```

Thank you for the answer, I appreciate it a lot but what I meant wasn't about the blade side. But in mypage class... Which classes I need to use/extend so that I can present that $form into the blade... Since filament is relatively new, I can't find it in google, lol...

I'm not that experienced in php and livewire but filament really helps me building something like a pro... 😁

**Reply by johncarter-** [reply]
This covers how to add forms to an existing Livewire component: https://filamentphp.com/docs/2.x/forms/getting-started#preparing-your-livewire-component
